From dc05da9aad1ea26635fca93316e0f48c0966608c Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Tue, 20 Dec 2016 13:55:43 -0800
Subject: [PATCH 4002/4002] Fix build on Mountain Lion

MachTask.mm:437:18: error: no member named 'external_page_count' in 'vm_statistics64'
          vminfo.external_page_count - vminfo.purgeable_count;
          ~~~~~~ ^
1 error generated.

https://llvm.org/bugs/show_bug.cgi?id=31438

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 tools/debugserver/source/MacOSX/MachTask.mm | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git llvm_master/tools/lldb/tools/debugserver/source/MacOSX/MachTask.mm macports_master/tools/lldb/tools/debugserver/source/MacOSX/MachTask.mm
index bd7047ecd..eeda83d33 100644
--- llvm_master/tools/lldb/tools/debugserver/source/MacOSX/MachTask.mm
+++ macports_master/tools/lldb/tools/debugserver/source/MacOSX/MachTask.mm
@@ -24,6 +24,7 @@
 #import <sys/sysctl.h>
 
 #if defined(__APPLE__)
+#include <Availability.h>
 #include <pthread.h>
 #include <sched.h>
 #endif
@@ -429,7 +430,7 @@ profile_data_stream << "wired:" << vminfo.wire_count * pagesize << ';';
 profile_data_stream << "active:" << vminfo.active_count * pagesize << ';';
 profile_data_stream << "inactive:" << vminfo.inactive_count * pagesize << ';';
  */
-#if defined(HOST_VM_INFO64_COUNT)
+#if defined(HOST_VM_INFO64_COUNT) && (!defined(__APPLE__) || __MAC_OS_X_VERSION_MAX_ALLOWED >= 1090)
       // This mimicks Activity Monitor.
       uint64_t total_used_count =
           (physical_memory / pagesize) -
-- 
2.13.2

